src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -196,6 +198,80 @@ public void GenerateHeader()
             PopBlock();
         }
 
+
+        private class DelegateSignature
+        {
+            public string Signature { get; set; }
+            public bool SuppressWarning { get; set; }
+            public string CallingConvention { get; set; }
```

Comment:
Please use the higher level types of CallingConvention and Type and call ToString when needed instead of calling it in advance as you do now.